tests/python/unittest/test_evaluator_flush_l2_cache.py

```diff
@@ -0,0 +1,63 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import tvm
+from tvm import te
+from tvm.script import tir as T
+import tvm.testing
+import numpy as np
+
+
+@T.prim_func
+def matmul(a: T.handle, b: T.handle, c: T.handle) -> None:
+    A = T.match_buffer(a, [128, 128])
+    B = T.match_buffer(b, [128, 128])
+    C = T.match_buffer(c, [128, 128])
+    for i, j, k in T.grid(128, 128, 128):
+        with T.block("matmul"):
+            vi, vj, vk = T.axis.remap("SSR", [i, j, k])
+            with T.init():
+                C[vi, vj] = 0.0
+            C[vi, vj] = C[vi, vj] + A[vi, vk] * B[vk, vj]
+
+
+@tvm.testing.requires_cuda
+def test_evaluator_flush_l2_cache():
+    mod = tvm.IRModule.from_expr(matmul)
+    sch = tvm.tir.Schedule(mod)
+    blk = sch.get_block("matmul")
+    i, j, k = sch.get_loops(blk)
+    sch.bind(i, "blockIdx.x")
+    sch.bind(j, "threadIdx.x")
+    f = tvm.build(sch.mod["main"], target="cuda")
+    dev = tvm.cuda(0)
+    evaluator_no_flush = f.time_evaluator(f.entry_name, dev, repeat=1000, number=1)
+
+    a = tvm.nd.array(np.random.rand(128, 128).astype("float32"), device=dev)
+    b = tvm.nd.array(np.random.rand(128, 128).astype("float32"), device=dev)
+    c = tvm.nd.array(np.zeros((128, 128)).astype("float32"), device=dev)
+    args = [a, b, c]
+    print("Evaluator (w/o L2 flush):\t{:.5f}ms".format(evaluator_no_flush(*args).mean * 1000))
+
+    evaluator_with_flush = f.time_evaluator(
+        f.entry_name, dev, repeat=1000, number=1, f_preproc="l2_cache_flush_cuda"
+    )
+    print("Evaluator (w/ L2 flush):\t{:.5f}ms".format(evaluator_with_flush(*args).mean * 1000))
```

Comment:
Are there any criteria where the test should be failed?

Reply:
There isn't but I'd love to have one, any ideas on that?

Reply:
> Are there any criteria where the test should be failed?

This test tests for the presence of **l2_cache_flush_cuda**. It's normal integration test :)

Not sure if we need to do the first run: evaluator_no_flush.
In some tests, the base scenario is hidden with an **if False** block.
For example you can look at do_tune block in  _tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py::test_packed_8x8x32_resnet50_
Or use **@pytest.mark.parametrize** for **f_preproc**

Reply:
`pytest.mark.parametrize` sounds good to me

Reply:
> This test tests for the presence of **l2_cache_flush_cuda**. It's normal integration test :)

Yes, but I thought if we can check that `l2_cache_flush_cuda` works. Previously, it was two runs of `time_evaluator`. And I thought if we can compare the execution time, then we probably can say if `l2_cache_flush_cuda` works or not.

Probably the better test for such functionality is to check L2 cache after `Flush` function. But probably such test should be implemented by using gtest, and I'm not sure that such infrastructure exists for cuda. Anyway, I don't think that we should spend a lot of time on implementing such test in this PR and I don't want to block this PR only because of such test. The existing test must be enough.